config/bool.go

```diff
@@ -0,0 +1,97 @@
+package config
+
+import (
+	"fmt"
+	"strings"
+)
+
+// GetBoolT returns the value associated with the given key as a boolean.
+//
+// If the value is "true", "yes", or "on", it returns true. If the value is
+// "false", "no", or "off", it returns false. These string comparisons are
+// case-insensitive. It returns an error if any other value is used.
+//
+// If they key is not defined, it returns true.
+func GetBoolT(b Bucket, k string) (bool, error) {
+	return GetBoolDefault(b, k, true)
+}
+
+// GetBoolF returns the value associated with the given key as a boolean.
+//
+// If the value is "true", "yes", or "on", it returns true. If the value is
+// "false", "no", or "off", it returns false. These string comparisons are
+// case-insensitive. It returns an error if any other value is used.
+//
+// If they key is not defined, it returns false.
+func GetBoolF(b Bucket, k string) (bool, error) {
+	return GetBoolDefault(b, k, false)
+}
+
+// GetBoolDefault returns the value associated with the given key as a
+// boolean.
+//
+// If the value is "true", "yes", or "on", it returns true. If the value is
+// "false", "no", or "off", it returns false. These string comparisons are
+// case-insensitive. It returns an error if any other value is used.
+//
+// If they key is not defined, it returns v.
+func GetBoolDefault(b Bucket, k string, v bool) (bool, error) {
+	x := b.Get(k)
+
+	if x.IsZero() {
+		return v, nil
+	}
+
+	s, err := x.AsString()
+	if err != nil {
+		return false, err
+	}
+
+	switch strings.ToLower(s) {
+	case "true", "yes", "on":
```

Comment:
Should these include `"1"`, and the falsy ones below include `"0"`?

Reply:
I was about to write my comment on this but then i saw @koden-km's comment. I feel like the boolean value extraction should deal with `1` and `0` as well. For instance `DEBUG_LOGGING=1` looks quite idiomatic.

Reply:
I had them in there originally then removed them, just to be more explicit (I also had `t`, `f`, `y` and `n`).

My logic is that if you're using this you're generally going to be writing new software so it doesn't need to be compatible with existing environment variables. But, I could be convinced to include them again.

Reply:
I thought `1` and `0` were the most common for boolean flags. I don't think I've ever seen `t`, `f`, `y` or `n` used before. so I wouldn't worry about those. The `true`/`false`, `on`/`off`, `yes`/`no` are clearer for everyone though and you wouldn't mistake those for integer values.

Reply:
I tend to agree, of all one-character values `0` and `1` definitely deserve staying.

Reply:
I'm not sure any of us agree :P  It sounds like Kev is arguing to allow `1` and `0`, and Danil is saying to allow all of the single-character variants? Do I have that right?


> I thought `1` and `0` were the most common for boolean flags

This *is* definitely common, but I can't actually think for a good argument why. Perhaps it's because in bash, et al, where you deal directly with strings you can compare it to `0` to check for truthiness, but we have a real `bool` type here in Go.

Reply:
If its all for new software and the docs are clear, then i'm fine with the full word booleans and ignoring the single letter and `0` and `1`.

Reply:
If nothing else we can always add support for these "shortcut" values later if we find we really want them, but it would be a BC break to add them now then take them away.

Reply:
My opinion is to leave `1` and `0` pair in, but to disallow any other one-character values; though, I am flexible with retaining only `true/false`, `on/off`, and `yes/no` thus adding clarity to the new software that will include `dodeca`.